projects/build-env/src/plugin/targets/environment.targets.ts

```diff
@@ -0,0 +1,145 @@
+    return false;
+  }
+
+  if (
+    existingTargetNames.some((existingTarget) =>
+      environmentTargetNames.includes(existingTarget)
+    )
+  ) {
+    if (existingTags && environmentsTagFilters) {
+      return existingTags.some((existingTag) =>
+        environmentsTagFilters.includes(existingTag)
+      );
+    }
+    return true;
+  }
+
+  return false;
+}
+
+export function verdaccioTargets(
+  projectConfig: ProjectConfiguration,
+  options: Pick<
+    NormalizedCreateNodeOptions['environments'],
+    'environmentsDir'
+  > &
+    Omit<StartVerdaccioOptions, 'projectName'>
+): Record<string, TargetConfiguration> {
+  const { name: envProject } = projectConfig;
+  const { environmentsDir, ...verdaccioOptions } = options;
+  const environmentDir = join(environmentsDir, envProject);
+
+  return {
+    [DEFAULT_START_VERDACCIO_TARGET]: {
+      // @TODO: consider using the executor function directly to reduce the number of targets
+      executor: '@nx/js:verdaccio',
+      options: {
+        config: '.verdaccio/config.yml',
+        port: uniquePort(),
+        storage: join(environmentDir, 'storage'),
+        clear: true,
+        environmentDir,
+        projectName: envProject,
+        ...verdaccioOptions,
+      },
+    },
+    [DEFAULT_STOP_VERDACCIO_TARGET]: {
+      executor: '@push-based/build-env:kill-process',
+      options: {
+        filePath: join(environmentsDir, VERDACCIO_REGISTRY_JSON),
+        ...verdaccioOptions,
+      },
+    },
+  };
+}
+
+export function getEnvTargets(
+  projectConfig: ProjectConfiguration,
+  options: NormalizedCreateNodeOptions['environments']
+): Record<string, TargetConfiguration> {
+  const { name: envProject } = projectConfig;
+  const { environmentsDir } = options;
+  const environmentRoot = join(environmentsDir, envProject);
+  return {
+    [DEFAULT_BOOTSTRAP_TARGET]: {
+      executor: '@push-based/build-env:bootstrap',
+      options: { environmentRoot },
+    },
+    // just here to execute dependent npm-install tasks with the correct environmentProject
+    [DEFAULT_INSTALL_TARGET]: {
+      dependsOn: [
+        {
+          projects: 'dependencies',
+          target: DEFAULT_NPM_INSTALL_TARGET,
+          params: 'forward',
+        },
+      ],
+      options: { environmentRoot },
+    },
+    // runs bootstrap-env, install-env and stop-verdaccio
+    [DEFAULT_SETUP_TARGET]: {
+      outputs: ['{options.environmentRoot}'],
+      executor: '@push-based/build-env:setup',
+      options: {
+        environmentRoot,
+      },
+    },
+  };
+}
+
+export function updateEnvTargetNames(
+  projectConfig: ProjectConfiguration,
+  options: Required<Pick<BuildEnvEnvironmentsOptions, 'targetNames'>>
+): Record<string, TargetConfiguration> {
+  const { targetNames: envTargetNames } = options;
+  const { targets: existingTargets = {} as TargetConfiguration } =
+    projectConfig;
+
+  return Object.fromEntries(
+    Object.entries(existingTargets).map(([existingTargetName, config]) => {
```

Comment:
maybe that's acceptable to use intermediary variables to make it more readable?

Reply:
The function `isEnvProject` is used to check if `updateEnvTargetNames` should execute. Is this what you ask for?

Reply:
Ah now I get it!! I can try

Reply:
So..

I created a helper:

```ts
export function iterateEntries<T, K extends string, V>(
  obj: T,
  cb: (entries: [keyof T, T[keyof T]][]) => [K, V][]
): { [key in K]: V } {
  return Object.fromEntries(
    cb(Object.entries(obj) as [keyof T, T[keyof T]][])
  ) as { [key in K]: V };
}
```

and use it like this:
```ts

  return iterateEntries(existingTargets, (entries) => entries
    .map(([existingTargetName, config]) => {
        // ...
        return [existingTargetName, config];
    })
  );
```